### DISCUSSION TITLE
Upload files to component related to an User model entry

### DISCUSSION THREAD

**Discussion by Developer-Anjan** [strapi/strapi#7985] [Legacy Topics]
**The problem**
I am trying to upload a file to strapi and trying to link it to an attribute of a nested component via **/upload** endpoint. Uploading is working well but linking is not happening. It is also not showing any error. 

I am trying this example but with a component
**https://strapi.io/documentation/v3.x/plugins/upload.html#upload-files**

so I am not sure what will be the value for the **field**  parameter.

**Steps to reproduce the behavior**

1. Create a component called **like** with a text field called **ip**
2. Create another component called **photo** and add a **media** field called **media**
3. add **like** component as _repeatable_ in the **photo** component.
4. Create the final component called **gallery** and add the **photo** component to it as repeatable.
5. Finally add **gallery** component to **users** collection as a single component.

The gallery component will be like this

![Screenshot (76)](https://user-images.githubusercontent.com/48857974/93867204-5790b380-fce6-11ea-9d43-3973c5457db1.png)

**Expected behavior**
Upload the image and associate it to the media field inside the photo component of user model.

**Additional context**
So I am trying to achieve the following: -

1. users will have a gallery.
2. gallery will have media and every media file will have comments and like.

so, is my approach correct ! or there is another way to achieve it.

**System**

- npm v6.14.7
- node v12.16.2
- strapi v3.1.5
- sqlite3 latest
- os win10

**Comment by derrickmehaffy** [comment]
Can you please open this as a bug report and not a discussion?

**Reply by Developer-Anjan** [reply]
But I don't know whether my approach is correct or not.

**Comment by alexandrebodin** [comment]
Hi, the `field` property would be equal to the path of the component inside the contentType. In your case somethin like:

`gallery.Photo[index].media`

**Reply by Developer-Anjan** [reply]
I tried that way also but it is not working. It doesn't even show any error if I enter a non-existing path also. Can you please recreate it and show me the outcome?

**Here is my request:**

![Screenshot (78)](https://user-images.githubusercontent.com/48857974/94031160-301c1280-fddc-11ea-8497-3ed004b689db.png)

**Here is response**

```json
[
    {
        "id": 67,
        "name": "big smaple.jpg",
        "alternativeText": null,
        "caption": null,
        "width": 817,
        "height": 505,
        "formats": {
            "thumbnail": {
                "name": "thumbnail_big smaple.jpg",
                "hash": "thumbnail_big_smaple_ec34e66b4d",
                "ext": ".jpg",
                "mime": "image/jpeg",
                "width": 245,
                "height": 151,
                "size": 6.42,
                "path": null,
                "url": "/uploads/thumbnail_big_smaple_ec34e66b4d.jpg"
            },
            "medium": {
                "name": "medium_big smaple.jpg",
                "hash": "medium_big_smaple_ec34e66b4d",
                "ext": ".jpg",
                "mime": "image/jpeg",
                "width": 750,
                "height": 464,
                "size": 44.31,
                "path": null,
                "url": "/uploads/medium_big_smaple_ec34e66b4d.jpg"
            },
            "small": {
                "name": "small_big smaple.jpg",
                "hash": "small_big_smaple_ec34e66b4d",
                "ext": ".jpg",
                "mime": "image/jpeg",
                "width": 500,
                "height": 309,
                "size": 22,
                "path": null,
                "url": "/uploads/small_big_smaple_ec34e66b4d.jpg"
            }
        },
        "hash": "big_smaple_ec34e66b4d",
        "ext": ".jpg",
        "mime": "image/jpeg",
        "size": 53.15,
        "url": "/uploads/big_smaple_ec34e66b4d.jpg",
        "previewUrl": null,
        "provider": "local",
        "provider_metadata": null,
        "created_by": null,
        "updated_by": null,
        "created_at": "2020-09-23T14:57:30.951Z",
        "updated_at": "2020-09-23T14:57:30.951Z",
        "related": [
            {
                "__contentType": "UsersPermissionsUser",
                "id": 15,
                "username": "user-2",
                "email": "user2@test.com",
                "provider": "local",
                "password": "$2a$10$.kj4XuT5Xs/tE2bEzlcIe.ne5BGLeZifbXJx1IW2ZxTnqV8r6ghh6",
                "resetPasswordToken": null,
                "confirmed": true,
                "blocked": null,
                "role": 1,
                "mobile": null,
                "phone": null,
                "facebook": null,
                "instagram": null,
                "linkedin": null,
                "website": null,
                "about": null,
                "address": null,
                "created_by": null,
                "updated_by": null,
                "created_at": "2020-09-22T05:51:28.065Z",
                "updated_at": "2020-09-22T05:51:28.171Z"
            }
        ]
    }
]
```

**Reply by Developer-Anjan** [reply]
@alexandrebodin Have you got any solution

**Reply by Rapido102** [reply]
up plz

**Reply by alexandrebodin** [reply]
If you upload the image inside a component the ref should point to the component not the main contentType :) in your example you need to get the component id and upload to the media field of that component directly. 

```
ref: category.component
id: component.id
```

**Reply by Rapido102** [reply]
Hi @alexandrebodin,

```
Card: Array(3)
0: {_id: "5fd731bf785c6c0446d6dea6", __v: 0, image: {…}, …}
1: {_id: "5fd731bf785c6c0446d6dea7", __v: 0, image: {…}, …}
2: {_id: "5fd731bf785c6c0446d6dea8", __v: 0, image: {…}, …}
length: 3
__proto__: Array(0)
boutonTitre: ""
boutonUrl: ""
content: ""
createdAt: "2020-10-30T13:28:05.488Z"
id: "5f9c14e5a2546227dc970d34" 

const updateImgCard = async (value, index) => {
        try {
            const formData = new FormData();
            formData.append('files', value[0]);
            formData.append('ref', 'homepage-content.Card');
            formData.append('refId', props.data.Card[index].id);
            formData.append('field', `image`);
            await axios.post(`${urle}/upload`, formData);
        } catch (e) {
            createNotification('error', 'Error message', e);
        }
    };
```
that's tell me "Cannot read property 'associations' of undefined", have you an idea ?

**Reply by alexandrebodin** [reply]
I think your ref might be invalid can you set it in lowercase ?

**Reply by Rapido102** [reply]
Same error.
